translate docs/create-a-new-react-app (#14)
reactjs/ja.reactjs.org: content/docs/create-a-new-react-app.md

---
id: create-a-new-react-app
title: 新しい React アプリを作る
permalink: docs/create-a-new-react-app.html
redirect_from:
  - "docs/add-react-to-a-new-app.html"
prev: add-react-to-a-website.html
next: cdn-links.html
---

もっとも良いユーザ・開発体験を得るために統合されたツールチェインを使いましょう。

このページではいくつかの人気のある React ツールチェインを説明します。これは次のようなタスクに役立ちます：

> * 大量のファイルとコンポーネントにスケールする

Reviewer: ```suggestion
* 大量のファイルとコンポーネントでスケールする
```

「大量のコンポーネントにスケール」と言うとコンポーネントが増えるのが嬉しいみたいに聞こえる気がします

Author: ありがとうございます！
あれ、コンポーネントが増やせる＝分割できるのが嬉しいみたいな認識でした…

Reviewer: たぶん「大量のコンポーネントがあっても大丈夫」の方だと思います

* npm を通してサードパーティライブラリを利用する
* よくある間違いを早期に発見する
* 開発環境で CSS と JS をライブ編集する
* 本番用の出力を最適化する

> このページで推奨されているツールチェインは**始めるに当たって必ずしも必要な設定ではありません**。

Reviewer: ```suggestion
このページで推奨されているツールチェインは**始めるにあたって設定が不要です**。
```

日本語だけ読んでたらすんなり意味が通りそうでしたが英文で言っていることと違うような…

Author: 確かに全く読み間違えてました…サジェスト適応します！

## ツールチェインが必要ない場合

> あなたが上記のような問題を経験していなかったり、まだ JavasScript のツールを利用するのに慣れていない場合、[HTML ページに簡単な `<script>` タグで React を追加](/docs/add-react-to-a-website.html)したり必要に応じて[JSX と利用](/docs/add-react-to-a-website.html#optional-try-react-with-jsx)することを検討してください。

Reviewer: ```suggestion
あなたが上記のような問題を経験していなかったり、まだ JavasScript のツールを利用するのに慣れていない場合、（必要に応じて[JSX と併用](/docs/add-react-to-a-website.html#optional-try-react-with-jsx)して）[HTML ページに単純な `<script>` タグで React を追加](/docs/add-react-to-a-website.html)することを検討してください。
```

原文に近づけました

Author: JSX は副次的な内容なので前の [adding React ...] より後ろに出てきた方が私はわかりやすいかなと思います。
うーん、`〜することを検討してみてください。（JSX の利用も検討してみてください）`というのはいかがでしょうか？

Reviewer: いいと思います

Author: ```suggestion
あなたが上記のような問題を経験していなかったり、まだ JavasScript のツールを利用するのに慣れていない場合、[HTML ページに簡単な `<script>` タグで React を追加](/docs/add-react-to-a-website.html)することを検討してください。（[JSX の利用](/docs/add-react-to-a-website.html#optional-try-react-with-jsx)も検討してみてください）
```

これは**既存のウェブサイトに React を統合する最も簡単な方法**でもあります。あなたが役立つと思えばいつでもより大きなツールチェインを追加できます。

## 推奨するツールチェイン

React チームは主に以下のソリューションを推奨します：

> - **React を学習中**か**新しい[単ページ](/docs/glossary.html#single-page-application)アプリを作成**するなら、[Create React App](#create-react-app) を利用してください

Reviewer: ```suggestion
- **React を学習中**か、**新しい[シングルページ](/docs/glossary.html#single-page-application)アプリケーションを作成したい**場合、[Create React App](#create-react-app) を利用してください
```

SPAはもうカタカナ語として語彙化していると思うのでそちらを使いましょう

Author: ここはどっちが通りが良いか悩んだので、シングルページで通るならそちらで！

- **Node.js でサーバサイドでレンダーされたウェブサイト**を構築するなら、[Next.js](#nextjs) を試してください
- **静的なコンテンツ中心のウェブサイト**を構築するなら、[Gatsby](#gatsby) を試してください
- **コンポーネントライブラリ**の構築や**既存のコードベースへの統合**をするなら、[その他の柔軟なツールチェイン](#more-flexible-toolchains)を試してください

### Create React App

> [Create React App](http://github.com/facebookincubator/create-react-app) は **React を学習する**のに快適な環境であり、React で**新しい[単ページ](/docs/glossary.html#single-page-application)アプリケーション**を作成するのに最も良い方法です。

Reviewer: ```suggestion
[Create React App](http://github.com/facebookincubator/create-react-app) は **React を学習する**のに快適な環境であり、React で**新しい[シングルページ](/docs/glossary.html#single-page-application)アプリケーション**を作成するのに最も良い方法です。
```

Author: 上と同様ですね

> 開発環境をセットアップして最新の JavaScript の機能を使え、快適な開発体験が得られ、本番環境用の最適化をしてくれます。あなたのマシンに Node >= 6 及び npm >= 5.2 の環境が必要です。プロジェクトを作成するには次を実行します：

Reviewer: ```suggestion
開発環境をセットアップして最新の JavaScript の機能を使えるようにし、快適な開発体験を提供し、そして本番環境用の最適化を行います。あなたのマシンに Node >= 6 及び npm >= 5.2 の環境が必要です。プロジェクトを作成するには次を実行します：
```

Author: これはサジェストしていただいた内容の方が良いですね

```bash
npx create-react-app my-app
cd my-app
npm start
```

>注釈
>
>最初の行の `npx` は打ち間違いではありません -- これは [npm 5.2 から利用できるパッケージランナーツール](https://medium.com/@maybekatz/introducing-npx-an-npm-package-runner-55f7d4bd282b)です.

Create React App はバックエンドのロジックやデータベース接続は扱いません。フロントエンドのビルドパイプラインを構築するだけであり、バックエンドに関しては好きなものを組み合わせて使って構いません。内部では [Babel](http://babeljs.io/) と [webpack](https://webpack.js.org/) を利用していますが、それらについて知る必要はありません。

本番環境にデプロイする準備ができたら、`npm run build` を実行すれば、`build` フォルダ内に最適化されたアプリケーションのビルドが生成されます。Create React App の詳細については、[該当ツールの README](https://github.com/facebookincubator/create-react-app#create-react-app-) および [ユーザガイド](https://github.com/facebookincubator/create-react-app/blob/master/packages/react-scripts/template/README.md#table-of-contents) を参照してください。

### Next.js

> [Next.js](https://nextjs.org/) は React の**静的なアプリケーション・サーバでレンダーされるアプリケーション**の構築に人気のある軽量フレームワークです。すぐに使える**スタイルおよびルーティングのソリューション**を含み、サーバ環境として [Node.js](https://nodejs.org/) を利用することを想定しています。

Reviewer: ```suggestion
[Next.js](https://nextjs.org/) は React を使って**静的かつサーバサイドでレンダーされるアプリケーション**を構築する場合に人気のある軽量フレームワークです。すぐに使える**スタイルおよびルーティングのソリューション**を含み、サーバ環境として [Node.js](https://nodejs.org/) を利用することを想定しています。
```

Author: 前後はいただいた通りで直したいです
強調の中は静的なサイトとサーバサイドでレンダーされるアプリケーションのことを言っているんじゃないでしょうか

Next.js の[オフィシャルガイド](https://nextjs.org/learn/)を参照してください。

### Gatsby

> [Gatsby](https://www.gatsbyjs.org/) は React で**静的なウェブサイト**を作成するのに最も良い方法です。React コンポーネントを使用しながらも、事前にレンダーされた HTML と CSS を出力して最速のロード時間を保証します。

Reviewer: ```suggestion
[Gatsby](https://www.gatsbyjs.org/) は React で**静的なウェブサイト**を作成するのに最も良い方法です。React コンポーネントを使用しながらも、事前レンダーされた HTML と CSS を出力することで最速のロード時間を保証します。
```

Author: サジェストいただいた方が良いのでそちらで！

Gatsby の[オフィシャルガイド](https://www.gatsbyjs.org/docs/) および [スターターキットのギャラリー](https://www.gatsbyjs.org/docs/gatsby-starters/)を参照してください。

### その他の柔軟なツールチェイン

> 次のツールチェインはより柔軟で選択肢があります。経験豊富なユーザにこれらを推奨します。

Reviewer: ```suggestion
以下のツールチェインはより大きな柔軟性や選択肢を提供します。経験豊富なユーザにこれらを推奨します。
```

ちょっと固いなら「以下のツールチェインを使えばより大きな柔軟性や選択肢が得られます」でもいいかも

> - **[Neutrino](https://neutrinojs.org/)** は [webpack](https://webpack.js.org/) のパワーとプリセットのシンプルさを兼ね備えています。プリセットには [React アプリ](https://neutrinojs.org/packages/react/) と [React コンポーネント](https://neutrinojs.org/packages/react-components/)があります。

Reviewer: ```suggestion
- **[Neutrino](https://neutrinojs.org/)** は [webpack](https://webpack.js.org/) のパワーとプリセットのシンプルさを兼ね備えています。プリセットには [React アプリ](https://neutrinojs.org/packages/react/) と [React コンポーネント](https://neutrinojs.org/packages/react-components/)用のものがあります。
```

Author: いただいた通りで直します

- **[nwb](https://github.com/insin/nwb)** は [npm で React コンポーネントを公開する](https://github.com/insin/nwb/blob/master/docs/guides/ReactComponents.md#developing-react-components-and-libraries-with-nwb)ことに特に優れています。React アプリを作成するのにも[利用](https://github.com/insin/nwb/blob/master/docs/guides/ReactApps.md#developing-react-apps-with-nwb)できます。 

> - **[Parcel](https://parceljs.org/)** は高速な、ゼロ設定のウェブアプリケーションバンドラで [React で動作](https://parceljs.org/recipes.html#react)します。

Reviewer: ```suggestion
- **[Parcel](https://parceljs.org/)** は高速な、ゼロ設定のウェブアプリケーションバンドラであり、[React と共に利用](https://parceljs.org/recipes.html#react)できます。
```

「Reactで動作します」だとPercel自体がReactベースで開発されているかのような印象を与えそうです

> - **[Razzle](https://github.com/jaredpalmer/razzle)** はその他の設定不要なサーバレンダリングフレームワークでありながら、Next.js よりの柔軟性があります。

Reviewer: ```suggestion
- **[Razzle](https://github.com/jaredpalmer/razzle)** は設定不要のサーバレンダリングフレームワークでありながら、Next.js よりも柔軟性があります。
```

Author: 誤字と読み間違いですね…サジェスト適応します！
